lib/cppcheck.cpp

```diff
@@ -310,7 +310,7 @@ static std::vector<picojson::value> executeAddon(const AddonInfo &addonInfo,
             //std::cout << "addon '" << addonInfo.name <<  "' result is not a JSON" << std::endl;
 
             result.erase(result.find_last_not_of('\n') + 1, std::string::npos); // Remove trailing newlines
-            throw InternalError(nullptr, "Failed to execute '" + pythonExe + " " + args + "'. " + result);
+            throw InternalError(nullptr, "Failed to execute '" + pythonExe + " " + args + "'. Expected '{' but got '" + line[0] + "' as first character.", "Output was: '" + result + "'");
```

Comment:
Okay so we should have written "line" instead of "result" here.

This modification of InternalError means that internal errors will not have separate short and verbose messages. I am skeptic about that. All other error messages have independant short and verbose messages.

Could we fix the bug here and replace the result with line. I am not against that we add a verbose output that contains all results but it would be good to point out the bad line..

short message:

    Failed to execute '...'. Invalid line: abcd

verbose message:

    Failed to execute '...'. Expected '{' but got 'a' as first character. Invalid line: abcd
    Results:
     ....

Reply:
Sounds good to not include all of the result but just the offending line. However in line with the `InternalError` logic I would suggest the following:

The throwing:
```
throw InternalError(nullptr,
   "Unexpected output from '" + pythonExe + " " + args + "'. Invalid line '"+line+"'",
   "Expected '{' but got '" + line[0] + "' as first character."
);
```
(I changed "failed to execute" because it actually did execute - we just don't like what it produced.)

The resulting short message:
```
Unexpected output from '...'. Invalid line 'abcd'
```

The resulting verbose message:
```
Unexpected output from '...'. Invalid line 'abcd': Expected '{' but got 'a' as first character. 
```

Still then, the question is what to do with `InternalError` messages containing `\n`? Or can we be confident that they never occur again..?

Reply:
> Still then, the question is what to do with InternalError messages containing \n? Or can we be confident that they never occur again..?

maybe your approach is fine.. however I do feel that there should really not be newlines.. ideally I would like to see some assertion error or internal error message that just complains that there is a newline..

Reply:
> (I changed "failed to execute" because it actually did execute - we just don't like what it produced.)

I like that.

Reply:
> > Still then, the question is what to do with InternalError messages containing \n? Or can we be confident that they never occur again..?
> 
> maybe your approach is fine.. however I do feel that there should really not be newlines.. ideally I would like to see some assertion error or internal error message that just complains that there is a newline..

I kind of agree, but imagine getting "an internal error occurred processing an internal error" - technically very accurate, but helpful... not really. Not for a new user, not for the bug report they will submit. Compile-time options seem very limited, short from enforce the short message of `InternalError` to be a string literal, and putting all dynamic/formatted/concatenated stuff in the details/verbose field. Can we simply take the first N characters while replacing `\n` with spaces?

Reply:
if a developer is debugging cppcheck I think a assertion error that terminates would be good.. but:
 1. we don't currently have assertions that only terminates during debugging.. I guess we would need to update the build scripts to set the NDEBUG macro or something..
 2. what if the user runs into the problem instead. we need a solution that is acceptable for the user.

> Can we simply take the first N characters while replacing \n with spaces?

sure